# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2005/01/29 14:26:17-02:00 james4765@cwazy.co.uk 
#   [PATCH] lcd: fix memory leak in lcd_ioctl()
#   
#   This patch fixes a memory leak in the FLASH_Burn ioctl for the Cobalt LCD interface driver.
#   
#   Signed-off-by: James Nelson <james4765@gmail.com>
# 
# drivers/char/lcd.c
#   2005/01/27 21:49:38-02:00 james4765@cwazy.co.uk +3 -1
#   lcd: fix memory leak in lcd_ioctl()
# 
diff -Nru a/drivers/char/lcd.c b/drivers/char/lcd.c
--- a/drivers/char/lcd.c	2005-02-03 05:58:23 -08:00
+++ b/drivers/char/lcd.c	2005-02-03 05:58:23 -08:00
@@ -438,8 +438,10 @@
 		save_flags(flags);
 		for (i=0; i<FLASH_SIZE; i=i+128) {
 
-		        if(copy_from_user(rom, display.RomImage + i, 128))
+		        if(copy_from_user(rom, display.RomImage + i, 128)) {
+			   kfree(rom);
 			   return -EFAULT;
+			}
 			burn_addr = kFlashBase + i;
 			cli();
 			for ( index = 0; index < ( 128 ) ; index++ ) 
